ts/sagas/wallet.ts

```diff
@@ -24,10 +93,217 @@ function* fetchTransactions(
+    );
+  } else {
+    yield put(paymentTransactionSummaryFromBanner());
+  }
+  // also, show summary screen
+  yield put(navigateTo(ROUTES.PAYMENT_TRANSACTION_SUMMARY));
+}
+
+function* continueWithPaymentMethodsHandler(
+  _: PaymentRequestContinueWithPaymentMethods
+) {
+  // find out whether a payment method has already
+  // been defined as favorite. If so, use it and
+  // ask the user to confirm it
+  // Otherwise, show a list of payment methods available
+  // TODO: if no payment method is available (or if the
+  // user chooses to do so), allow adding a new one.
+  const favoriteCard: Option<number> = yield select(getFavoriteWalletId);
+  if (favoriteCard.isSome()) {
+    // show card
+    yield put(paymentRequestConfirmPaymentMethod(favoriteCard.value));
+  } else {
+    // show list
+    yield put(paymentRequestPickPaymentMethod());
+  }
+}
+
+function* confirmPaymentMethodHandler(
+  action: PaymentRequestConfirmPaymentMethod
+) {
+  const walletId = action.payload;
+  yield put(paymentConfirmPaymentMethod(walletId));
+  yield put(navigateTo(ROUTES.PAYMENT_CONFIRM_PAYMENT_METHOD));
+}
+
+function* pickPaymentMethodHandler(_: PaymentRequestPickPaymentMethod) {
+  // show screen with list of payment methods available
+  yield put(paymentPickPaymentMethod());
+  yield put(navigateTo(ROUTES.PAYMENT_PICK_PAYMENT_METHOD));
+}
+
+function* completionHandler(_: PaymentRequestCompletion) {
+  // -> it should proceed with the required operations
+  // and terminate with the "new payment" screen
+
+  // do payment stuff (-> pagoPA REST)
+  // retrieve transaction and store it
+  // mocked data here
+  const wallet: Wallet = yield select(selectedPaymentMethodSelector);
+  // tslint:disable-next-line saga-yield-return-type
+  const recipient = (yield select(getPaymentRecipient)).getOrElse(
+    UNKNOWN_RECIPIENT
+  );
+  // tslint:disable-next-line saga-yield-return-type
+  const paymentReason = (yield select(getPaymentReason)).getOrElse(
+    UNKNOWN_PAYMENT_REASON
+  );
+  const fee = feeExtractor(wallet);
+
+  const transaction: WalletTransaction = {
+    id: Math.floor(Math.random() * 1000),
+    cardId: wallet.idWallet === undefined ? -1 : wallet.idWallet,
+    isoDatetime: new Date().toISOString(),
+    paymentReason:
+      paymentReason === undefined ? UNKNOWN_PAYMENT_REASON : paymentReason,
+    recipient: (recipient === undefined ? UNKNOWN_RECIPIENT : recipient)
+      .denominazioneBeneficiario,
+    amount: AmountInEuroCentsFromNumber.encode(
+      yield select(getCurrentAmount) // tslint:disable-line saga-yield-return-type
+    ),
+    currency: "EUR",
+    transactionCost: AmountInEuroCentsFromNumber.encode(
+      fee === undefined ? UNKNOWN_AMOUNT : fee
+    ),
+    isNew: true
+  };
+  yield put(storeNewTransaction(transaction));
+  yield put(selectTransactionForDetails(transaction));
+  yield put(selectWalletForDetails(getWalletId(wallet))); // for the banner
+  yield put(
+    // TODO: this should use StackActions.reset
```

Comment:
is there a story for this TODO?

Reply:
There is, I forgot to mention [it](https://www.pivotaltracker.com/story/show/159300579) in the code